internal/server/data/organization.go

```diff
@@ -14,36 +14,37 @@ func CreateOrganization(tx GormTxn, org *models.Organization) error {
 		return fmt.Errorf("creating org: %w", err)
 	}
 
-	tx = NewTransaction(tx.GormDB(), org.ID)
-
-	_, err = initializeSettings(tx)
+	_, err = initializeSettings(tx, org.ID)
 	if err != nil {
 		return fmt.Errorf("initializing org settings: %w", err)
 	}
 
 	infraProvider := &models.Provider{
-		Name:      models.InternalInfraProviderName,
-		Kind:      models.ProviderKindInfra,
-		CreatedBy: models.CreatedBySystem,
+		Name:               models.InternalInfraProviderName,
+		Kind:               models.ProviderKindInfra,
+		CreatedBy:          models.CreatedBySystem,
+		OrganizationMember: models.OrganizationMember{OrganizationID: org.ID},
 	}
 	if err := CreateProvider(tx, infraProvider); err != nil {
 		return fmt.Errorf("failed to create infra provider: %w", err)
 	}
 
 	connector := &models.Identity{
-		Name:      models.InternalInfraConnectorIdentityName,
-		CreatedBy: models.CreatedBySystem,
+		Name:               models.InternalInfraConnectorIdentityName,
+		CreatedBy:          models.CreatedBySystem,
+		OrganizationMember: models.OrganizationMember{OrganizationID: org.ID},
 	}
 	// this identity is used to create access keys for connectors
 	if err := CreateIdentity(tx, connector); err != nil {
 		return fmt.Errorf("failed to create connector identity while creating org: %w", err)
 	}
 
 	err = CreateGrant(tx, &models.Grant{
-		Subject:   uid.NewIdentityPolymorphicID(connector.ID),
-		Privilege: models.InfraAdminRole,
-		Resource:  "infra",
-		CreatedBy: models.CreatedBySystem,
+		Subject:            uid.NewIdentityPolymorphicID(connector.ID),
+		Privilege:          models.InfraAdminRole,
```

Comment:
```suggestion
		Privilege:          models.InfraConnectorRole,
```

This was fixed in my recent PR, but I want to make sure it doesn't regress here.